src/StoreApi/Routes/V1/CartUpdateCustomer.php

```diff
@@ -70,15 +113,28 @@ public function get_args() {
 	 */
 	protected function get_route_post_response( \WP_REST_Request $request ) {
 		$cart     = $this->cart_controller->get_cart_instance();
-		$billing  = $request['billing_address'] ?? [];
-		$shipping = $request['shipping_address'] ?? [];
 		$customer = wc()->customer;
 
+		// Get data from request object and merge with customer object.
+		$billing  = wp_parse_args( $request['billing_address'] ?? [], $this->get_customer_billing_address( $customer ) );
+		$shipping = wp_parse_args( $request['shipping_address'] ?? [], $this->get_customer_shipping_address( $customer ) );
```

Comment:
This code merges the request data, with the current customer data. It ensures that we have a basic set of address data (including the country) upon which validation relies, even if the data was not included in the request.

This allows pushes of partial addresses.